### PR TITLE
Enable querying cumulative metrics with agg_time_dimension

### LINKED ISSUE #1000: [SL-1604] [Feature] Enable querying cumulative metrics with their agg_time_dimension
### Is this your first time submitting a feature request?

- [X] I have read the [expectations for open source contributors](https://docs.getdbt.com/docs/contributing/oss-expectations)
- [X] I have searched the existing issues, and I could not find an existing issue for this feature
- [X] I am requesting a straightforward extension of existing metricflow functionality, rather than a Big Idea better suited to a discussion

### Describe the feature

Currently, cumulative metrics can only be queried with `metric_time`. Since metric_time represents the `agg_time_dimensions` associated with the metric's measures, you should be able to replace `metric_time` with the `agg_time_dimension` and get the same behavior.
Note: if a metric has multiple input measures with different `agg_time_dimensions`, `metric_time` will still be required.

### Describe alternatives you've considered

_No response_

### Who will this benefit?

_No response_

### Are you interested in contributing this feature?

_No response_

### Anything else?

_No response_

<sub>[SL-1604](https://linear.app/dbt-labs/issue/SL-1604/[feature]-enable-querying-cumulative-metrics-with-their-agg-time)</sub>

### PR DESCRIPTION
**Pull request by courtneyholcomb**
Resolves #1000

### Description
Enable querying cumulative metrics with their `agg_time_dimension`. Should have the same behavior as when queried with `metric_time`.

### CODE REVIEW

File: metricflow/query/validation_rules/metric_time_requirements.py [hunk 1]
```diff
@@ -34,29 +33,11 @@ class MetricTimeQueryValidationRule(PostResolutionQueryValidationRule):
...
-        for time_granularity in TimeGranularity:
-            metric_time_specs.append(
-                TimeDimensionSpec(
-                    element_name=METRIC_TIME_ELEMENT_NAME,
-                    entity_links=(),
-                    time_granularity=time_granularity,
-                    date_part=None,
-                )
+        self._metric_time_specs = tuple(
+            TimeDimensionSpec.generate_possible_specs_for_time_dimension(
```
Comment (tlento) [inline metricflow/query/validation_rules/metric_time_requirements.py:37]: Nice.

**Review by tlento — COMMENTED** [review]
Looks good, thanks for splitting this up into such bite-sized chunks for review!

There are a few boundary cases I'd like to test here and read through before approving.

1. The time range constraint SQL rendering - the query runs, but I worry this is because the optimizer collapses the metric_time references into direct column name references, and it's not clear to me what the raw rendering will look like. I see this is tested for result correctness in the integration test (which is great) but there is no SQL to look at right now.
2. Do we render correctly for metrics that take multiple measures with the same agg_time_dimension?
3. Do we throw the right error for metrics that take multiple measures with different agg_time_dimensions?

File: metricflow/dataflow/builder/dataflow_plan_builder.py [hunk 2]
```diff
@@ -1356,11 +1370,13 @@ def _build_aggregated_measure_from_measure_source_node(
...
             unaggregated_measure_node = filtered_measure_source_node
 
+        # If time constraint was previously adjusted for cumulative window or grain, apply original time constraint
+        # here. Can skip if metric is being aggregated over all time.
         cumulative_metric_constrained_node: Optional[ConstrainTimeRangeNode] = None
         if (
             cumulative_metric_adjusted_time_constraint is not None
             and time_range_constraint is not None
-            and queried_linkable_specs.contains_metric_time
+            and queried_agg_time_dimension_specs
```
[the remark this answers is not in the document]
Comment (courtneyholcomb) [reply metricflow/dataflow/builder/dataflow_plan_builder.py:1379]: Yep I'm aware! But good point that I actually I should revert the change on this line so it only applies the time constraint with `metric_time` - forgot to do that.
I had actually put up a separate issue for this since it's not directly related to the feature: https://github.com/dbt-labs/metricflow/issues/1008
Since the time constraint param isn't exposed in any of our APIs, only in the MF CLI, it seems less important to fix that issue before releasing this feature.
Note: in the integration test you noted, I used a where filter instead of a time constraint.

[the remark this answers is not in the document]
Comment (courtneyholcomb) [reply metricflow/dataflow/builder/dataflow_plan_builder.py]: I just put up a commit to raise a more appropriate error if `metric_time` is not used here.

**Comment by courtneyholcomb** [comment]
> Do we render correctly for metrics that take multiple measures with the same agg_time_dimension?
> Do we throw the right error for metrics that take multiple measures with different agg_time_dimensions?

@tlento I realized this isn't relevant to this PR since cumulative metrics can't have multiple input measures. So I'll move dealing with this feedback to the time offset PR!

File: metricflow/dataflow/builder/dataflow_plan_builder.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (tlento) [reply metricflow/dataflow/builder/dataflow_plan_builder.py]: > Since the time constraint param isn't exposed in any of our APIs, only in the MF CLI, it seems less important to fix that issue before releasing this feature.

Right, makes sense, although we still don't want result divergence for people who do local dev with the CLI and then issue where filters in prod cloud accounts.

If I'm honest, though, I'm mostly looking ahead to predicate pushdown implementation concerns. :)

I'll take a look at the update!

**Review by tlento — APPROVED** [review]
LGTM, thanks!
